src/file-events/cpp/linux_fsnotifier.cpp

```diff
@@ -19,15 +19,22 @@ WatchPoint::WatchPoint(const u16string& path, shared_ptr<Inotify> inotify, int w
     , path(path) {
 }
 
-bool WatchPoint::cancel() {
+CancelResult WatchPoint::cancel() {
     if (status == WatchPointStatus::CANCELLED) {
-        return false;
+        return CancelResult::ALREADY_CANCELLED;
     }
     status = WatchPointStatus::CANCELLED;
     if (inotify_rm_watch(inotify->fd, watchDescriptor) != 0) {
-        throw FileWatcherException("Couldn't stop watching", path, errno);
+        switch (errno) {
+            case EINVAL:
+                logToJava(LogLevel::INFO, "Couldn't stop watching %s (probably because the directory was removed)", utf16ToUtf8String(path).c_str());
```

Comment:
I wonder if the INFO log level is too high...

Reply:
We do `INFO` for other unexpected events, too (like `Path is not watched`).